### PR TITLE
Add available volumes to binder index when not present

### LINKED ISSUE #9273: PVs where Phase=Available need reindexing
In case of cluster crash and status recovery, the persistentVolumeClaimBinder would not re-add available volumes to its internal index.  Users would not be able to claim those volumes.

@thockin

### PR DESCRIPTION
**Pull request by markturansky**
Resolves #9273

Short term fix is to add the PVs to the index.

Long term fix is to remove the Pending phase entirely.

unit test exposes the bug.

@thockin

### CODE REVIEW

File: pkg/volumeclaimbinder/persistent_volume_claim_binder.go [hunk 1]
```diff
@@ -158,6 +158,14 @@ func syncVolume(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCl
 
 	// available volumes await a claim
 	case api.VolumeAvailable:
+		// TODO:  remove api.VolumePending phase altogether
```
Comment (thockin) [inline pkg/volumeclaimbinder/persistent_volume_claim_binder.go]: @bgrant0607 re: API changes - how breaking is removing a phase in your eyes?  Should we press to get this in before 1.0?

Comment (thockin) [inline pkg/volumeclaimbinder/persistent_volume_claim_binder.go]: My feeling is "not very breaking" which means we could drop that phase after 1.0 without much impact

**Comment by thockin** [comment]
LGTM 627dd86

**Comment by bgrant0607** [comment]
Removing phases is less breaking than adding them.

**Pull request merged**